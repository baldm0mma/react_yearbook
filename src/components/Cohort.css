.allStaff {
  display: flex;
  flex-wrap: wrap;
}

.staff_header {
  background-color: aliceblue;
  color: blue;
}