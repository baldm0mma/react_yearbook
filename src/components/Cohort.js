import React from 'react';
import Person from './Person';
import './Cohort.css';

const Cohort = props => {
  const staffMembers = props.staff.map(member => {
    return (
      <Person
        key={member.id}
        name={member.name}
        quote={member.quote}
        superlative={member.superlative}
        photo={member.photo}
      />
    );
  });

  const students = props.students.map(member => {
    return (
      <Person
        key={member.id}
        name={member.name}
        quote={member.quote}
        superlative={member.superlative}
        photo={member.photo}
      />
    );
  });

  return (
    <div>
      <h2 className='staff_header'>Staff</h2>
      <div className='allStaff'>{staffMembers}</div>
      <h2 className='student_header'>Students</h2>
      <div className='allStudents'>{students}</div>
    </div>
  );
};

export default Cohort;
