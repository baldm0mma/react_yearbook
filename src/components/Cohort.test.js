import React from 'react';
import { shallow, mount } from 'enzyme';
import Cohort from './Cohort';

describe('Cohort', () => {
  it('should render the correct elements with the correct information', () => {
    const wrapper = shallow(<Cohort staff={ [] } students={ [] }/>);

    expect(wrapper).toMatchSnapshot();
  });
});