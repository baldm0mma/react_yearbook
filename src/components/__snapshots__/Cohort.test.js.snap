// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cohort should render the correct elements with the correct information 1`] = `
<div>
  <h2
    className="staff_header"
  >
    Staff
  </h2>
  <div
    className="allStaff"
  />
  <h2
    className="student_header"
  >
    Students
  </h2>
  <div
    className="allStudents"
  />
</div>
`;
